ultraplot/axes/geo.py

```diff
@@ -761,10 +778,80 @@ def format(
                 nsteps=nsteps,
             )
+        # Set tick lengths for flat projections
+        if lonticklen or latticklen:
+            # Only add warning when ticks are given
+            if _is_rectilinear_projection(self):
+                self._add_geoticks("x", lonticklen, ticklen)
+                self._add_geoticks("y", latticklen, ticklen)
+            else:
+                warnings._warn_ultraplot(
+                    f"Projection is not flat ignoring {lonticklen=} and {latticklen=} settings"
+                )
 
         # Parent format method
         super().format(rc_kw=rc_kw, rc_mode=rc_mode, **kwargs)
 
+    def _add_geoticks(self, x_or_y, itick, ticklen):
+        """
+        Add tick marks to the geographic axes.
+
+        Parameters
+        ----------
+        x_or_y : {'x', 'y'}
+            The axis to add ticks to ('x' for longitude, 'y' for latitude).
+        itick, tick : unit-spec, default: :rc:`tick.len`
+            Major tick lengths for the x and y axis.
+            %(units.pt)s
+            Use the keyword `ticklen` to set both at once.
+        Notes
+        -----
+        This method handles proper tick mark drawing for geographic projections
+        while respecting the current gridline settings.
+        """
+
+        size = _not_none(itick, ticklen)
+        # Skip if no tick size specified
+        if size is None:
+            return
+        size = units(size)
+
+        ax = getattr(self, f"{x_or_y}axis")
+
+        # Get the tick positions based on the locator
+        gl = self.gridlines_major
+        # Note: set_xticks points to a different method than self.[x/y]axis.set_ticks from the mpl backend. For basemap we are adding the ticks to the mpl backend and for cartopy we are simple using their functions by showing the axis.
+        if isinstance(gl, tuple):
+            locator = gl[0] if x_or_y == "x" else gl[1]
+            tick_positions = np.asarray(list(locator.keys()))
+            # Show the ticks but hide the labels
+            ax.set_ticks(tick_positions)
+            ax.set_major_formatter(mticker.NullFormatter())
+
+        # Always show the ticks
+        ax.set_visible(True)
+
+        # Apply tick parameters
+        # Move the labels outwards if specified
+        # Offset of 2 * size is aesthetically nice
+        if isinstance(gl, tuple):
+            locator = gl[0] if x_or_y == "x" else gl[1]
+            for loc, objects in locator.items():
+                for object in objects:
+                    # text is wrapped in a list
+                    if isinstance(object, list) and len(object) > 0:
+                        object = object[0]
+                    if isinstance(object, mtext.Text):
+                        object.set_visible(True)
+        else:
+            setattr(gl, f"{x_or_y}padding", 2 * size)
+
+        # Note: set grid_alpha to 0 as it is controlled through the gridlines_major object (which is not the same ticker)
```

Comment:
Am I understanding correctly that a user cannot use both ticks and grid lines? We should leave that as configurable by the user and not force them. A nice default is good of course. As the code looks now, I think folks would have to add the gridlines back after adding ticks. Instead, they should only have to make a single call to `.format` that adds ticks and sets the gridline alpha to zero.

Reply:
The ticks are exaggerated in the tests to allow the users to change the size of the ticks

Reply:
I am confused then. Why is the core setting the grid alpha to zero when adding ticks then?

Reply:
otherwise it doubles the grid lines.

Reply:
Geoaxes are a bit strange. The set tick functions are also no properly overwritten (issue with cartopy) the grid lines in UltraPlot are set manually in a separate function and set tick Params would add it independently